client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -638,6 +638,10 @@ public CompletableFuture<Boolean> createStream(String scope, String streamName,
             case INVALID_STREAM_NAME:
                 log.warn(requestId, "Illegal stream name: {}", streamName);
                 throw new IllegalArgumentException("Illegal stream name: " + streamConfig);
+
+            case INVALID_RETENTION_POLICY:
+                log.warn(requestId, "Invalid retention policy for the stream: {}", streamName);
+                throw new IllegalArgumentException("Invalid retention policy as require retention policy is enabled: " + streamConfig);
```

Comment:
Message could be simplified/improved.
something like this "Missing retention policy. Retention policy must be specified."

Reply:
> Message could be simplified/improved. something like this "Missing retention policy. Retention policy must be specified."

Should we also specify when this retention policy is mandatory, something think like below.
"Missing retention policy. Retention policy is mandatory if 'requireRetentionPolicy' flag is enabled".

Reply:
in case of `null` also the stream creation will fail now, updated the condition